torcs_ros_client/src/torcs_ros_client_node.cpp

```diff
@@ -178,22 +182,24 @@ std::string TORCSROSClient::sensorsMsgToString(){
   result += SimpleParser::stringify("track", track_array_, config_.num_track_ranges);
   result += SimpleParser::stringify("trackPos", (float) torcs_sensors_.trackPos);
   result += SimpleParser::stringify("wheelSpinVel", wheelSpinVel_, 4);
-  result += SimpleParser::stringify("z", (float)torcs_global_.pose.position.z);
-  result += SimpleParser::stringify("x", (float)torcs_global_.pose.position.x); 
-  result += SimpleParser::stringify("y", (float)torcs_global_.pose.position.y); 
-  result += SimpleParser::stringify("roll", (float)torcs_global_.roll);
-  result += SimpleParser::stringify("pitch", (float)torcs_global_.pitch); 
-  result += SimpleParser::stringify("yaw", (float)torcs_global_.yaw); 
-  result += SimpleParser::stringify("speedGlobalX", (float)torcs_global_.speedGlobal.linear.x); 
-  result += SimpleParser::stringify("speedGlobalY", (float)torcs_global_.speedGlobal.linear.y); 
+  result += SimpleParser::stringify("z", (float)globalPose_.pose.position.z);
+  result += SimpleParser::stringify("x", (float)globalPose_.pose.position.x); 
+  result += SimpleParser::stringify("y", (float)globalPose_.pose.position.y); 
+  result += SimpleParser::stringify("roll", (float)globalRPY_.vector.x);
+  result += SimpleParser::stringify("pitch", (float)globalRPY_.vector.y); 
+  result += SimpleParser::stringify("yaw", (float)globalRPY_.vector.z); 
+  result += SimpleParser::stringify("speedGlobalX", (float)globalSpeed_.twist.linear.x); 
+  result += SimpleParser::stringify("speedGlobalY", (float)globalSpeed_.twist.linear.y); 
 
 
 
   return result;
 }
 void TORCSROSClient::sensorsMsgFromString(std::string torcs_string){
   torcs_sensors_.header.stamp = ros::Time::now();
-  torcs_global_.header.stamp = ros::Time::now();
+  globalSpeed_.header.stamp = ros::Time::now();
+  globalPose_.header.stamp = ros::Time::now();
```

Comment:
Please also add the frame_id (`globalSpeed_.header.frame_id="world"`) to the header of the global pose

Reply:
frame_id has been added to all newly introduced geometry messages (speed, pose and rpy)